airbyte-integrations/connectors/source-e2e-test/src/main/java/io/airbyte/integrations/source/e2e_test/ContinuousFeedConfig.java

```diff
@@ -0,0 +1,203 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class ContinuousFeedConfig {
+
+  private static final JsonNode JSON_SCHEMA_DRAFT_07;
+  private static final JsonSchemaValidator SCHEMA_VALIDATOR = new JsonSchemaValidator();
+  private static final ObjectMapper MAPPER = MoreMappers.initMapper();
+
+  static {
+    try {
+      final String jsonSchemaDraft07 = MoreResources.readResource("json_schema_draft_07.json");
+      JSON_SCHEMA_DRAFT_07 = Jsons.deserialize(jsonSchemaDraft07);
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public enum MockCatalogType {
+    SINGLE_STREAM,
+    MULTI_STREAM
+  }
+
+  private final long seed;
+  private final AirbyteCatalog mockCatalog;
+  private final long maxMessages;
+  private final Optional<Long> messageIntervalMs;
+
+  public ContinuousFeedConfig(final JsonNode config) throws JsonValidationException {
+    this.seed = parseSeed(config);
+    this.mockCatalog = parseMockCatalog(config);
+    this.maxMessages = parseMaxMessages(config);
+    this.messageIntervalMs = parseMessageIntervalMs(config);
+  }
+
+  static long parseSeed(final JsonNode config) {
+    if (!config.has("seed")) {
+      return System.currentTimeMillis();
+    }
+    return config.get("seed").asLong();
+  }
+
+  static AirbyteCatalog parseMockCatalog(final JsonNode config) throws JsonValidationException {
+    final JsonNode mockCatalogConfig = config.get("mock_catalog");
+    final MockCatalogType mockCatalogType = MockCatalogType.valueOf(mockCatalogConfig.get("type").asText());
+    switch (mockCatalogType) {
+      case SINGLE_STREAM -> {
+        final String streamName = mockCatalogConfig.get("stream_name").asText();
+        final String streamSchemaText = mockCatalogConfig.get("stream_schema").asText();
+        final Optional<JsonNode> streamSchema = Jsons.tryDeserialize(streamSchemaText);
+        if (streamSchema.isEmpty()) {
+          throw new JsonValidationException(String.format("Stream \"%s\" has invalid schema: %s", streamName, streamSchemaText));
+        }
+        processSchema(streamSchema.get());
+        checkSchema(streamName, streamSchema.get());
+
+        final AirbyteStream stream = new AirbyteStream().withName(streamName).withJsonSchema(streamSchema.get());
+        return new AirbyteCatalog().withStreams(Collections.singletonList(stream));
+      }
+      case MULTI_STREAM -> {
+        final String streamSchemasText = mockCatalogConfig.get("stream_schemas").asText();
+        final Optional<JsonNode> streamSchemas = Jsons.tryDeserialize(streamSchemasText);
+        if (streamSchemas.isEmpty()) {
+          throw new JsonValidationException("Input stream schemas are invalid: %s" + streamSchemasText);
+        }
+
+        final List<AirbyteStream> streams = new LinkedList<>();
```

Comment:
This list is only going to be traversed linearly. We don't need to access an element with a certain index in the middle. I think a linked list is most suitable for this use case.

The downside of an array list is that the list may be re-sized when new elements are added to it, which is tiny bit less efficient than linked list.

Reply:
ah lol. the impression that I've gotten is that in practice, arraylists are often still faster (because they take better advantage of memory locality). but realistically I don't think it matters either way here 👍 

(though also I think the optimal implementation would be storing `MoreIterators.toList(streamSchemas.get().fields())`  into a variable and then using `new ArrayList(streamSchemaList.size())` to avoid the array resize)

Reply:
Good point. I will change it.